backend/game/engine/player_queue.py

```diff
@@ -0,0 +1,45 @@
+import itertools
+
+
+class GamePlayers():
+
+    def __init__(self):
+        self.all_players = []
+        self.dead_players = []
+        self.alive_players = []
+        self.current_player = None
+
+    def _add_initial_player(self, player):
+        self.all_players.append(player)
+
+    def _remove_initial_player(self, player):
+        if player in self.all_players:
+            self.all_players.remove(player)
+
+    def set_player_order(self):
+        # TODO: set this based off of dice roll minigame
+        # Currently just the order in which they were added to the list
+        self.alive_players = self.all_players
+        self.player_cycle = itertools.cycle(self.all_players)
+
+    def get_current_player(self):
+        if not self.current_player:
+            self.get_next_player()
+        return self.current_player
+
+    def get_next_player(self):
+        self.current_player = self._cycle_next_alive_player()
+
+    def get_alive_player(self):
+        return self.alive_players
+
+    def get_all_other_alive_player(self):
+        return [player for player in self.alive_players if
+                self.current_player != player]
+
+    def _cycle_next_alive_player(self):
+        for player in self.player_cycle:
```

Comment:
Where is this player_cycle? Why wouldn't you just use `self.players`?

Reply:
The cycle is in line 31, once the game begins and the player list/order is set, a cycle is created so that we do not need to worry about indexes or positions or dealing with the list type in a way that goes against python best pratices, we can just iterate through the cycle.